src/Microsoft.Azure.WebJobs.Host/Bindings/BindingContext.cs

```diff
@@ -83,6 +86,18 @@ public TraceWriter Trace
         }
 
         /// <summary>
+        /// Gets the set of property values for the current bind operation.
+        /// </summary>
+        /// <remarks>
+        /// Bindings can add arbitrary values to this collection as needed to
+        /// flow state around.
+        /// </remarks>
+        public IDictionary<string, object> Properties
```

Comment:
BindingData is coming from BindingContext.BindingData and contains the actual invoke values from the request (e.g. what our default resolver uses to resolve parameters from incoming binding data). Open to naming suggestions for "Properties". It's just a bag for arbitrary use.